pkg/kubelet/container_bridge.go

```diff
@@ -122,24 +140,26 @@ func cbr0Exists() (bool, error) {
 	return true, nil
 }
 
-func cbr0CidrCorrect(wantCIDR *net.IPNet) bool {
-	output, err := exec.Command("ip", "addr", "show", "cbr0").Output()
-	if err != nil {
+func cbr0CidrCorrect(br *netlink.Bridge, wantCIDR *net.IPNet) bool {
+
+	addrs, err := netlink.AddrList(br, syscall.AF_INET)
+	if err != nil && err != syscall.ENOENT {
+		glog.Errorf("could not get list of IP addresses from bridge %q: %v", br.Name, err)
 		return false
 	}
-	match := cidrRegexp.FindSubmatch(output)
-	if len(match) < 2 {
+
+	// TODO: Document why two addresses are required, and why the second one is chosen
+	if len(addrs) < 2 {
 		return false
 	}
-	cbr0IP, cbr0CIDR, err := net.ParseCIDR(string(match[1]))
+	_, cbr0CIDR, err := net.ParseCIDR(addrs[1].IPNet.String())
```

Comment:
The code should just loop through all addresses on  the bridge and look for the one we want.  If it finds it, return true.  If not, return false.